### ISSUE TITLE
Ensure CLI args override on-disk config

### ISSUE THREAD

**Issue by michaelsproul** [sigp/lighthouse#905]
## Description

I realised I was running my node with `--dummy-eth1`, so I stopped it, restarted it with `--eth1 --eth1-endpoint $URL`. I expected the node to use the CLI flags, but it still reported:

```
Mar 10 23:49:02.800 WARN Block production impaired               reason: dummy eth1 backend is enabled
```

Moving the `~/.lighthouse/beacon/beacon-node.toml` file out of the way and restarting got it to recognise the CLI options.

## Expected Behaviour

The CLI arguments should override any on-disk values at runtime. We don't yet want to try overwriting the on-disk values with the CLI flags, because the user may have inserted comments in their config (and comment-preserving editing is left for future work). We _do_ want to ensure that if the user provides a CLI flag, it takes effect and runtime and isn't ignored.

**Comment by michaelsproul** [comment]
Having some end-to-end tests to catch regressions would also be good.

**Comment by FawadHa1der** [comment]
Taking a stab at it

**Comment by FawadHa1der** [comment]
@michaelsproul I could not reproduce this one. I did this 
Steps
1) Started lighthouse beacon --dummy-eth1 and then Ctrl C to terminate

and then 
2) lighthouse beacon --eth1 --http --eth1-endpoint "https://goerli.infura.io/v3/XXXXXX"

Sorry for the very late reply, was adjusting to the new realities around.

**Comment by michaelsproul** [comment]
Interesting, which branch are you using? I just reproduced it on the machine where I initially saw the bug, but it was using a somewhat old variant of the `testnet5` branch. I'm rebuilding it on `master` now.

**Comment by FawadHa1der** [comment]
Yep on master.

**Comment by michaelsproul** [comment]
I just reproduced it on master. I suspect the issue might be that your on-disk config has `dummy-eth1` disabled. Can you try:

```
rm -r ~/.lighthouse/beacon
lighthouse beacon --dummy-eth1
lighthouse beacon --eth1 --http --eth1-endpoint "https://goerli.infura.io/v3/XXXXXX"
```

For me, those exact steps yield: `WARN Block production impaired               reason: dummy eth1 backend is enabled`

**Comment by michaelsproul** [comment]
Optionally, back up `~/.lighthouse/beacon/beacon-node.toml` before you delete it (it'll have your config options in there)

**Comment by FawadHa1der** [comment]
yep got it, thanks.

**Comment by AgeManning** [comment]
We've updated the logic for the CLI and I've made some modifications in v0.2.0. 

I think this has been resolved and #992 updates the CLI use. 

Can re-open if I am mistaken
